### ISSUE TITLE
Cannot import Pod framework that has other framework dependencies in UITests

### ISSUE THREAD

**Issue by sciutand** [CocoaPods/CocoaPods#4638]
Here a sample project
https://github.com/sciutand/SampleProjectThatFails.git

I'm running 
- Xcode 7.1
- cocoapod 0.39.0

For simplicity reason in the sample project the Kit is added to the iOS app directly instead of using a podspec but the outcome is the same.

The goal is to have a shared library (SampleKit), an iOS application that depends on the SampleKit and UITests for the iOS application.
All work fine until I try to import and use the SampleKit inside the UITests target. No matter what I do I always get a `library not loaded on Alamofire`:

```
XCTRunner[53330:4433829] The bundle “SampleProjectThatFailsUITests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2015-12-08 17:47:44.005 XCTRunner[53330:4433829] (dlopen_preflight(/Users/andrea/Library/Developer/CoreSimulator/Devices/9CFC265C-24E6-429E-9D36-FE3E25ECB4C9/data/Containers/Bundle/Application/38CFB479-7470-4A66-B3C9-1F31F705ECA1/SampleProjectThatFailsUITests-Runner.app/PlugIns/SampleProjectThatFailsUITests.xctest/SampleProjectThatFailsUITests): Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /Users/andrea/Library/Developer/CoreSimulator/Devices/9CFC265C-24E6-429E-9D36-FE3E25ECB4C9/data/Containers/Bundle/Application/38CFB479-7470-4A66-B3C9-1F31F705ECA1/SampleProjectThatFailsUITests-Runner.app/PlugIns/SampleProjectThatFailsUITests.xctest/SampleProjectThatFailsUITests
```

I'm not sure what the correct set up or import should be.

Any help greatly appreciated.

**Comment by segiddins** [comment]
This is likely the same issue as https://github.com/CocoaPods/CocoaPods/issues/4609. I'm guessing the UI Test target will need to have it's framework search paths set correctly by CocoaPods, but I'm not certain.

**Comment by sciutand** [comment]
> I'm guessing the UI Test target will need to have it's framework search paths set correctly by CocoaPods

Is it something I can test manually? Like what search paths should I set. I noticed in `...frameworks.sh` generated by cocoa for the Embed frameworks run script
> #Embed linked Swift runtime libraries. No longer necessary as of Xcode 7.
  if [ "${XCODE_VERSION_MAJOR}" -lt 7 ]; then

wouldn't this actually stop the SampleKit frameworks being added and therefore available to subsequent dependencies?

**Comment by segiddins** [comment]
Nope, that's just the swift runtime dylibs and is unrelated.

**Comment by sciutand** [comment]
Framework Search Paths resolves to $(inherited) and $PODS_FRAMEWORK_BUILD_PATH"

> build/Debug-iphoneos/Pods-SampleProjectThatFailsUiTests

which indeed has the `Alamofire.framework` inside it.

If I copy and paste the `Alamofire.framework` to `build/Debug-iphoneos/` the UITests target then runs fine.

**Comment by segiddins** [comment]
Super weird. Since you shared a project though, anyone can try and debug, so thanks :D

**Comment by sciutand** [comment]
Yes weird indeed. This problem definitely affects only `UI Testing bundle` target type, `Unit Testing Bundle` target type are unaffected.

On more thing, Cocoapod is not adding as `Embed Pods Frameworks` build phase for the `SampleProjectThatFailsUITests` unlike other Tests targets. I did add one manually but it makes no difference.
Still not sure if this is an unresolvable xcode7 issue with UITest targets or there is something I can add manually to temporary patch it, other than adding a build script that copies the frameworks to the root build folder

**Comment by sciutand** [comment]
:-( wrote the script that copies the pods framework where xcode is happy to find them but now facing this:
```
: Library not loaded: @rpath/libswiftCoreData.dylib
  Referenced from:
```

**Comment by sciutand** [comment]
After much looking around I believe this might be a voluntary xcode decision not to include external frameworks in the UITest bundle. Since the testBundle runs on a separate process there is no way you could share data or singletons between application and testBundle. The testBundle should be purely about testing the UI via the XCUI.. classes and API. I was hoping to somehow manage application state (such as logged in logged out, data displayed etc) but it looks like I'll have to keep pressing buttons to get where I want to be inside the app.

**Comment by segiddins** [comment]
Thanks for all the digging! Let's definitely leave it open for now

**Comment by neonichu** [comment]
This issue has been inactive for a long time and will be closed because we want to move forward with a clean slate after our 1.0 release. If you believe this is still relevant, please retest with the 1.0 release candidate and comment with a reproducible project in order for this to become actionable again. Thanks!

**Comment by ManueGE** [comment]
I still have this issue with 1.0 when I am trying to build a **dynamic library**. [Here is a simple project](https://www.dropbox.com/s/5vn8sjz2t6h3u2s/test_library_project.zip?dl=0) with a single pod to test. I can build it, but can't launch the tests. I am using Xcode 7.3.1.

It is working ok if I build a iOS app instead of a library. 

Thanks.

**Comment by segiddins** [comment]
@ManueGE that's intended, CocoaPods does not yet support integrating into library targets.

**Comment by ManueGE** [comment]
If anyone is interested, I fixed my problem with the library replacing my old Podfile.

It was:
````
platform :ios, '8.0'
use_frameworks!

target 'test' do
	pod 'Alamofire', '~> 3.4'
    
    target 'testTests' do
        inherit! :search_paths
    end
end
````

And I changed to
````
platform :ios, '8.0'
use_frameworks!

target 'test' do
	pod 'Alamofire', '~> 3.4'
end

target 'testTests' do
    pod 'Alamofire', '~> 3.4'
end
````

or using abstract targets:
````
platform :ios, '8.0'
use_frameworks!

abstract_target 'Base' do
	pod 'Alamofire', '~> 3.4'

    target 'test'
    target 'testTests'
end
````

**Comment by nap-sam-dean** [comment]
Hi,

I'm having the same issue with cocoapods 1.0.1 - my UI tests target is missing an embed frameworks phase. I can fix the issue by manually copying the phase from the Unit test target and changing the script names and it then works fine.

**Comment by abfelix** [comment]
I am also encountering this issue with cocoapods version 1.0.1. Also has missing embed frameworks in my UI test target. So what I did is add a run script in UI Test target Build Phase that imports the frameworks. The test then started to run but failed with this error:

Assertion Failure: UI Testing Failure - Failure getting snapshot Error Domain=XCTestManagerErrorDomain Code=9 "Error -25201 getting snapshot for element <AXUIElement 0x7f8659e0e6a0> {pid=40808}" UserInfo={NSLocalizedDescription=Error -25201 getting snapshot for element <AXUIElement 0x7f8659e0e6a0> {pid=40808}}

**Comment by bfolkens** [comment]
@nap-sam-dean @abfelix that fixed it for me too

**Comment by xun91** [comment]
Huge Thanks to @nap-sam-dean @abfelix 

I was having the same issue and I solved it by adding **Run Script** to the UITests Target Build Phase.
Using: cocoapods v1.0.0
`"${SRCROOT}/Pods/Target Support Files/Pods-PROJECTTests/Pods-PROJECTUITests-frameworks.sh"
`

**Comment by dral3x** [comment]
Same here. Fixed with @nap-sam-dean tip

**Comment by markm** [comment]
Thanks @nap-sam-dean - worked for me too

**Comment by DanToml** [comment]
Embedding frameworks in UI tests is also fixed on current master by #5503
